docs/architecture.md

```diff
@@ -1,188 +1,260 @@
 
-`policy-server` will exit with an error if one or more policies received wrong
-configuration parameters from the end user.
+You use policy configuration parameters to tune a policies' behavior.
+After startup and policy download the `policy-server`
+checks the policy settings provided by the user are valid.
 
-If all the policies are properly configured, `policy-server` will spawn a
-pool of worker threads to evaluate incoming requests using the Kubewarden
-policies specified by the user.
+The `policy-server` validates policy settings by invoking
+the `validate_setting` function exposed by each policy.
+There is further documentation in the
+[writing policies section](/writing-policies/spec/01-intro-spec.md)
+of the documentation.
 
-Finally, `policy-server` will start a HTTPS server that listens to incoming
-validation requests. The web server is secured using the TLS key and certificate
-that have been previously created by `kubewarden-controller`.
+<!--TODO:Next paragraph. 'end-user'? How are the configuration parameters
+specified for the `policy-server`-->
 
-Each policy is exposed by the web server via a dedicated path that follows this
-naming convention: `/validate/<policy ID>`.
+The `policy-server` exits with an error if one or more policies
+received wrong configuration parameters from the end-user.
 
-This is how the cluster looks like once the initialization of `policy-server`
-is completed:
+<!--TODO:Can we show the pool of worker threads in the diagram? Or maybe
+it's the stacked things?-->
+
+If all policies are correctly configured,
+`policy-server`
+spawns a pool of worker threads
+to evaluate incoming requests
+using the Kubewarden policies
+specified by the user.
+
+<!--TODO:Is the HTTPS server referred to, that which is in the aquamarine
+box, the 'Kubewarden Policy Server-->
+
+Finally, the `policy-server` starts a HTTPS server
+listening to incoming validation requests.
+Kubewarden uses the TLS key and certificate
+created by `kubewarden-controller`
+to secure the web server.
+
+The web server exposes each policy by a dedicated path
+following the naming convention: `/validate/<policy ID>`.
+
+This diagram shows the cluster when initialization of `policy-server` is complete:
 
 ![policy-server initialized](/img/architecture_sequence_03.png)
 
 ### Making Kubernetes aware of the policy
 
+<!--TODO:Next paragraph. Says 'admission reviews', should that maybe be
+'admission requests'-->
+
 The `policy-server` Pods have a
 [`Readiness Probe`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/),
-`kubewarden-controller` relies on that to know when the `policy-server` Deployment
-is ready to evaluate admission reviews.
+that `kubewarden-controller` uses to check when
+the `policy-server` Deployment is ready to evaluate admission reviews.
+
+<!--TODO:Aware? Is notified? Notifies? Is 'notifies' technically correct? Messages?-->
```

Comment:
The "notifies" here means that the controller will register the `MutatingWebhookConfiguration` or `ValidatingWebhookConfiguration` in the Kubernetes control plane to tell it that it must ask Kubewarden policies if a new resource is allowed in the cluster or not.

Reply:
The `*WebhookConfiguration` objects is the way you tell Kubernetes that it must ask a 3rd party to validate a new resource in the cluster.

Reply:
ok, thanks